src/modules/billing/mappers/charge-module-decorators.js

```diff
@@ -44,25 +45,30 @@ const decorateTransaction = (serviceTransaction, cmTransaction) =>
     isMinimumCharge: cmTransaction.minimumChargeAdjustment
   });
 
+const decorateMinimumChargeTransaction = async cmTransaction => {
+  const { transaction } = await chargeModuleTransactionConnector.get(cmTransaction.id);
+  return mappers.transaction.cmToModel(transaction);
+};
+
 /**
  * Maps all CM transaction data to the invoice licence
  * @param {InvoiceLicence} invoiceLicence containing service transactions
  * @param {Array} cmTransactions for that given licence
  */
-const decorateTransactions = (invoiceLicence, cmTransactions) => {
+const decorateTransactions = async (invoiceLicence, cmTransactions) => {
   if (isEmpty(invoiceLicence.transactions)) return invoiceLicence;
-
-  cmTransactions.map(cmTransaction => {
+  for (const cmTransaction of cmTransactions) {
     const serviceTransaction = invoiceLicence.transactions.find(trans => trans.externalId === cmTransaction.id);
     if (serviceTransaction) {
-      return decorateTransaction(serviceTransaction, cmTransaction);
-    }
-    if (!cmTransaction.minimumChargeAdjustment) {
+      decorateTransaction(serviceTransaction, cmTransaction);
+    } else if (!cmTransaction.minimumChargeAdjustment) {
+      // the only transactions which do not exist in the service should be minimum charge
       throw new TransactionStatusError(`Unexpected Charge Module transaction externalId: ${cmTransaction.id}`);
```

Comment:
Thanks for walking me through this. Shall we just move the error throwing to the end of the loop? It's not mandatory - Just good convention I think.

Reply:
Yes, I think that it would make it a tad bit clearer as well :)